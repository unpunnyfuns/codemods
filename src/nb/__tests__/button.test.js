import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/button', () => {
  it('migrates Button with leftIcon and children to nordlys Button', () => {
    const output = testTransform('nb/button', 'nb/button', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Button',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
