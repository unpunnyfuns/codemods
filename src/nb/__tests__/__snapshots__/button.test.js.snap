// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/button > migrates Button with leftIcon and children to nordlys Button 1`] = `
"import { useTranslation } from 'react-i18next'

import { Button } from '@target/components/Button';
import { View, StyleSheet } from 'react-native';
import { space } from '@design-tokens';

export function MyComponent() {
  const { t } = useTranslation()
  const text = 'Submit'

  return (
    <>

      {/* Dropped during migration (not supported in React Native):
        leftIcon={...} */
      }
      <View style={styles.button0}>
        <Button
          testID="button1"
          onPress={() => {}}
          size={"md"}
          icon={'PlusOutlined'}
          text={t('addNew')}
          variant='secondary'
          type='solid' />
      </View>
      <Button
        onPress={() => {}}
        isDisabled
        size={"lg"}
        icon={undefined}
        text={'Simple text'}
        variant='primary'
        type='solid' />
      <Button
        onPress={() => {}}
        size={"md"}
        icon={undefined}
        text={text}
        variant='secondary'
        type='solid' />
    </>
  );
}

const styles = StyleSheet.create({
  button0: {
    marginTop: space['2xl']
  }
});
"
`;
